### DISCUSSION TITLE
Manage streams / KVs across multiplea accounts with one user

### DISCUSSION THREAD

**Discussion by pcsegal** [nats-io/nats-server#5606] [Q&A] [answered]
When NATS is configured with multiple accounts, I can have one user for each account and, if I need to manage streams / KVs in an account, I need to authenticate as the user of the respective account.
However, it looks like I can't have a single user that can manage streams/KVs in multiple accounts.
I have a scenario where I would like to manage streams in multiple accounts. Is there a way of doing that without having to separately connect as each user's account?

**Comment by Jarema** [comment]
Each account is a separate tenant, with fully isolated subject context etc.
You cannot use one user to manage multiple streams.
Even if you want to share some data between accounts, you need to explicitly specify imports/exports.

You can however use system account to overview the whole NATS & JetStream system.

**Reply by brettinternet** [reply]
I didn't think the system account could manage the JetStream system. I'm getting this when I attempt to add unlimited limits to the system account.

```
[FTL] Not allowed to enable JetStream on the system account
```

**Reply by derekcollison** [reply, marked as answer]
The system account provides JetStream as a service to all enabled accounts. So that is correct, you can not enable JetStream on the system account.

**Comment by ripienaar** [comment]
You should be able to import the foreign account jetstream API and manage it using the API prefix options in clients and CLI

**Reply by brettinternet** [reply]
For a large multitenant system, is there a theoretical maximum number of imports for a single account? I'm imagining importing thousands of tenants for IoT user accounts into a "server client" in Go server that creates streams and publishes server side events to remote clients.

**Comment by atlejensen** [comment]
It depends on your usecase, but there are some options outside NATS.
Synadia Control Plane builds on top of your NATS environment(s) and gives you an overlay where you can manage multiple accounts with a single user. check out https://docs.synadia.com/platform/control-plane
There is also Qaze, a desktop tool that may solve some of your challenges. check out https://qaze.app/

**Comment by jnmoyne** [comment]
You can source streams from one account to another and have (some) control over what subjects get transferred: https://github.com/synadia-labs/cross-account-jetstream-sourcing
